pkg/controller/volumeattachment.go

```diff
@@ -86,20 +83,14 @@ func (r *ReconcileVolumeAttachment) Reconcile(ctx context.Context, request recon
 	}
 
 	azVolumeAttachmentName := azureutils.GetAzVolumeAttachmentName(diskName, volumeAttachment.Spec.NodeName)
-	azVolumeAttachment, err := azureutils.GetAzVolumeAttachment(ctx, r.controllerSharedState.cachedClient, r.controllerSharedState.azClient, azVolumeAttachmentName, r.controllerSharedState.objectNamespace, true)
-	if err != nil {
-		return reconcileReturnOnError(ctx, &volumeAttachment, "get azvolumeattachment", err, r.controllerRetryInfo)
-	}
 
-	updateFunc := func(obj client.Object) error {
-		azv := obj.(*azdiskv1beta2.AzVolumeAttachment)
-		// Update the annotation of AzVolumeAttachment with volumeAttachment name
-		azv.Status.Annotations = azureutils.AddToMap(azv.Status.Annotations, consts.VolumeAttachmentKey, volumeAttachment.Name)
-		return nil
-	}
-
-	if err = azureutils.UpdateCRIWithRetry(ctx, nil, r.controllerSharedState.cachedClient, r.controllerSharedState.azClient, azVolumeAttachment, updateFunc, consts.NormalUpdateMaxNetRetry, azureutils.UpdateCRIStatus); err != nil {
-		return reconcileReturnOnError(ctx, azVolumeAttachment, "update volumeattachment annotation", err, r.controllerRetryInfo)
+	if val, exists := r.controllerSharedState.azVolumeAttachmentToVaMap.Load(azVolumeAttachmentName); exists {
+		storedVolumeAttachmentName := val.(string)
+		if storedVolumeAttachmentName != volumeAttachment.Name {
+			r.controllerSharedState.azVolumeAttachmentToVaMap.Store(azVolumeAttachmentName, volumeAttachment.Name)
+		}
+	} else {
```

Comment:
Why not just unconditionally `Store` the `AzVolumeAttachment` name? If check for a name change seems unnecessary and shouldn't hurt to set it to the same name.